Amazon.Ion.ObjectMapper/IonListSerializer.cs

```diff
@@ -42,7 +45,22 @@ public override System.Collections.IList Deserialize(IIonReader reader)
                 var typedArray = Array.CreateInstance(elementType, list.Count);
                 for (int i=0; i<list.Count; i++)
                 {
-                    typedArray.SetValue(list[i], i);
+                    try
+                    {
+                        typedArray.SetValue(list[i], i);
+                    }
+                    catch (Exception e)
+                    {
+                        if (isPermissiveMode && (e is ArgumentException || e is InvalidCastException))
+                        {
```

Comment:
it is better to catch the specific exception, rather than testing for the exception type with if.

Reply:
From what I've read, C# [doesn't allow to catch multiple exceptions in one catch block](https://stackoverflow.com/questions/136035/catch-multiple-exceptions-at-once).

Reply:
There is a when keyword that you can use to pick the specific exceptions: 
catch (Exception e) when (e is ArgumentException || e is InvalidCastException)